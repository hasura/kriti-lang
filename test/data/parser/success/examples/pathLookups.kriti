# Path Lookups
[ {{ $foo }},
  {{ $foo.bar.baz }},
  {{ $foo['bar baz'].qux }},
  {{ $foo[1][0][3].ok }},
  {{ $foo?.bar }},
  {{ $foo?.['bar'] }},
  {{ $foo?.[1] }},
  {{ $foo?.[1]?.baz }}
]