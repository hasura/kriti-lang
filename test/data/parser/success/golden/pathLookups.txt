Array
    ( Span
        { start = AlexSourcePos
            { line = 1
            , col = 1
            }
        , end = AlexSourcePos
            { line = 9
            , col = 2
            }
        }
    )
    [ Path
        ( Span
            { start = AlexSourcePos
                { line = 1
                , col = 6
                }
            , end = AlexSourcePos
                { line = 1
                , col = 10
                }
            }
        )
        [ Obj
            ( Span
                { start = AlexSourcePos
                    { line = 1
                    , col = 6
                    }
                , end = AlexSourcePos
                    { line = 1
                    , col = 10
                    }
                }
            ) NotOptional "$foo" Head
        ]
    , Path
        ( Span
            { start = AlexSourcePos
                { line = 2
                , col = 6
                }
            , end = AlexSourcePos
                { line = 2
                , col = 18
                }
            }
        )
        [ Obj
            ( Span
                { start = AlexSourcePos
                    { line = 2
                    , col = 6
                    }
                , end = AlexSourcePos
                    { line = 2
                    , col = 10
                    }
                }
            ) NotOptional "$foo" Head
        , Obj
            ( Span
                { start = AlexSourcePos
                    { line = 2
                    , col = 10
                    }
                , end = AlexSourcePos
                    { line = 2
                    , col = 14
                    }
                }
            ) NotOptional "bar" DotAccess
        , Obj
            ( Span
                { start = AlexSourcePos
                    { line = 2
                    , col = 14
                    }
                , end = AlexSourcePos
                    { line = 2
                    , col = 18
                    }
                }
            ) NotOptional "baz" DotAccess
        ]
    , Path
        ( Span
            { start = AlexSourcePos
                { line = 3
                , col = 6
                }
            , end = AlexSourcePos
                { line = 3
                , col = 25
                }
            }
        )
        [ Obj
            ( Span
                { start = AlexSourcePos
                    { line = 3
                    , col = 6
                    }
                , end = AlexSourcePos
                    { line = 3
                    , col = 10
                    }
                }
            ) NotOptional "$foo" Head
        , Obj
            ( Span
                { start = AlexSourcePos
                    { line = 3
                    , col = 10
                    }
                , end = AlexSourcePos
                    { line = 3
                    , col = 21
                    }
                }
            ) NotOptional "bar baz" BracketAccess
        , Obj
            ( Span
                { start = AlexSourcePos
                    { line = 3
                    , col = 21
                    }
                , end = AlexSourcePos
                    { line = 3
                    , col = 25
                    }
                }
            ) NotOptional "qux" DotAccess
        ]
    , Path
        ( Span
            { start = AlexSourcePos
                { line = 4
                , col = 6
                }
            , end = AlexSourcePos
                { line = 4
                , col = 22
                }
            }
        )
        [ Obj
            ( Span
                { start = AlexSourcePos
                    { line = 4
                    , col = 6
                    }
                , end = AlexSourcePos
                    { line = 4
                    , col = 10
                    }
                }
            ) NotOptional "$foo" Head
        , Arr
            ( Span
                { start = AlexSourcePos
                    { line = 4
                    , col = 10
                    }
                , end = AlexSourcePos
                    { line = 4
                    , col = 13
                    }
                }
            ) NotOptional 1
        , Arr
            ( Span
                { start = AlexSourcePos
                    { line = 4
                    , col = 13
                    }
                , end = AlexSourcePos
                    { line = 4
                    , col = 16
                    }
                }
            ) NotOptional 0
        , Arr
            ( Span
                { start = AlexSourcePos
                    { line = 4
                    , col = 16
                    }
                , end = AlexSourcePos
                    { line = 4
                    , col = 19
                    }
                }
            ) NotOptional 3
        , Obj
            ( Span
                { start = AlexSourcePos
                    { line = 4
                    , col = 19
                    }
                , end = AlexSourcePos
                    { line = 4
                    , col = 22
                    }
                }
            ) NotOptional "ok" DotAccess
        ]
    , Path
        ( Span
            { start = AlexSourcePos
                { line = 5
                , col = 6
                }
            , end = AlexSourcePos
                { line = 5
                , col = 15
                }
            }
        )
        [ Obj
            ( Span
                { start = AlexSourcePos
                    { line = 5
                    , col = 6
                    }
                , end = AlexSourcePos
                    { line = 5
                    , col = 10
                    }
                }
            ) NotOptional "$foo" Head
        , Obj
            ( Span
                { start = AlexSourcePos
                    { line = 5
                    , col = 10
                    }
                , end = AlexSourcePos
                    { line = 5
                    , col = 15
                    }
                }
            ) Optional "bar" DotAccess
        ]
    , Path
        ( Span
            { start = AlexSourcePos
                { line = 6
                , col = 6
                }
            , end = AlexSourcePos
                { line = 6
                , col = 18
                }
            }
        )
        [ Obj
            ( Span
                { start = AlexSourcePos
                    { line = 6
                    , col = 6
                    }
                , end = AlexSourcePos
                    { line = 6
                    , col = 10
                    }
                }
            ) NotOptional "$foo" Head
        , Obj
            ( Span
                { start = AlexSourcePos
                    { line = 6
                    , col = 10
                    }
                , end = AlexSourcePos
                    { line = 6
                    , col = 18
                    }
                }
            ) Optional "bar" BracketAccess
        ]
    , Path
        ( Span
            { start = AlexSourcePos
                { line = 7
                , col = 6
                }
            , end = AlexSourcePos
                { line = 7
                , col = 14
                }
            }
        )
        [ Obj
            ( Span
                { start = AlexSourcePos
                    { line = 7
                    , col = 6
                    }
                , end = AlexSourcePos
                    { line = 7
                    , col = 10
                    }
                }
            ) NotOptional "$foo" Head
        , Arr
            ( Span
                { start = AlexSourcePos
                    { line = 7
                    , col = 10
                    }
                , end = AlexSourcePos
                    { line = 7
                    , col = 14
                    }
                }
            ) Optional 1
        ]
    , Path
        ( Span
            { start = AlexSourcePos
                { line = 8
                , col = 6
                }
            , end = AlexSourcePos
                { line = 8
                , col = 19
                }
            }
        )
        [ Obj
            ( Span
                { start = AlexSourcePos
                    { line = 8
                    , col = 6
                    }
                , end = AlexSourcePos
                    { line = 8
                    , col = 10
                    }
                }
            ) NotOptional "$foo" Head
        , Arr
            ( Span
                { start = AlexSourcePos
                    { line = 8
                    , col = 10
                    }
                , end = AlexSourcePos
                    { line = 8
                    , col = 14
                    }
                }
            ) Optional 1
        , Obj
            ( Span
                { start = AlexSourcePos
                    { line = 8
                    , col = 14
                    }
                , end = AlexSourcePos
                    { line = 8
                    , col = 19
                    }
                }
            ) Optional "baz" DotAccess
        ]
    ]